### ISSUE TITLE
Helm install fails due to CRD conflict in otherwise empty cluster

### ISSUE THREAD

**Issue by oivindoh** [containous/maesh#392]
Struggling with a strange issue installing maesh on a fresh k8s 1.17.0 cluster with helm3.0.1. Maesh chart repo updated prior to running install command. Helm fails with "Unable to continue with install: existing resource conflict: kind: CustomResourceDefinition, namespace: , name: traffictargets.access.smi-spec.io" - this CRD (or any other cmi-spec CRD) does not exist in the cluster prior to running helm install. 

```shell
$ helm version
version.BuildInfo{Version:"v3.0.1", GitCommit:"7c22ef9ce89e0ebeb7125ba2ebf7d421f3e82ffa", GitTreeState:"clean", GoVersion:"go1.13.4"}

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-13T11:52:32Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}

$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "maesh" chart repository
Update Complete. ⎈ Happy Helming!⎈
```

No smi-related crds are present in the cluster prior to installing. This is a 100% fresh kubeadm-deployed cluster with calico and traefik being the only additions as part of cluster standup:

```shell
$ kubectl get crd
NAME                                          CREATED AT
bgpconfigurations.crd.projectcalico.org       2019-12-20T13:20:41Z
bgppeers.crd.projectcalico.org                2019-12-20T13:20:41Z
blockaffinities.crd.projectcalico.org         2019-12-20T13:20:41Z
clusterinformations.crd.projectcalico.org     2019-12-20T13:20:41Z
felixconfigurations.crd.projectcalico.org     2019-12-20T13:20:41Z
globalnetworkpolicies.crd.projectcalico.org   2019-12-20T13:20:42Z
globalnetworksets.crd.projectcalico.org       2019-12-20T13:20:42Z
hostendpoints.crd.projectcalico.org           2019-12-20T13:20:42Z
ingressroutes.traefik.containo.us             2019-12-20T13:24:17Z
ingressroutetcps.traefik.containo.us          2019-12-20T13:24:17Z
ipamblocks.crd.projectcalico.org              2019-12-20T13:20:42Z
ipamconfigs.crd.projectcalico.org             2019-12-20T13:20:42Z
ipamhandles.crd.projectcalico.org             2019-12-20T13:20:42Z
ippools.crd.projectcalico.org                 2019-12-20T13:20:42Z
middlewares.traefik.containo.us               2019-12-20T13:24:17Z
networkpolicies.crd.projectcalico.org         2019-12-20T13:20:42Z
networksets.crd.projectcalico.org             2019-12-20T13:20:42Z
tlsoptions.traefik.containo.us                2019-12-20T13:24:17Z
```

Trying to install maesh as the first command after the previous results in failure:

```shell
$ helm install maesh maesh/maesh
Error: rendered manifests contain a resource that already exists. Unable to continue with install: existing resource conflict: kind: CustomResourceDefinition, namespace: , name: traffictargets.access.smi-spec.io
```

After the above runs, the trafficsplit and traffictargets crds are present in the cluster:

```shell
$ kubectl get crd
NAME                                          CREATED AT
bgpconfigurations.crd.projectcalico.org       2019-12-20T13:33:23Z
bgppeers.crd.projectcalico.org                2019-12-20T13:33:23Z
blockaffinities.crd.projectcalico.org         2019-12-20T13:33:24Z
clusterinformations.crd.projectcalico.org     2019-12-20T13:33:24Z
felixconfigurations.crd.projectcalico.org     2019-12-20T13:33:24Z
globalnetworkpolicies.crd.projectcalico.org   2019-12-20T13:33:24Z
globalnetworksets.crd.projectcalico.org       2019-12-20T13:33:24Z
hostendpoints.crd.projectcalico.org           2019-12-20T13:33:24Z
httproutegroups.specs.smi-spec.io             2019-12-20T13:49:11Z
ingressroutes.traefik.containo.us             2019-12-20T13:35:48Z
ingressroutetcps.traefik.containo.us          2019-12-20T13:35:48Z
ipamblocks.crd.projectcalico.org              2019-12-20T13:33:24Z
ipamconfigs.crd.projectcalico.org             2019-12-20T13:33:24Z
ipamhandles.crd.projectcalico.org             2019-12-20T13:33:24Z
ippools.crd.projectcalico.org                 2019-12-20T13:33:24Z
middlewares.traefik.containo.us               2019-12-20T13:35:48Z
networkpolicies.crd.projectcalico.org         2019-12-20T13:33:24Z
networksets.crd.projectcalico.org             2019-12-20T13:33:24Z
tcproutes.specs.smi-spec.io                   2019-12-20T13:49:11Z
tlsoptions.traefik.containo.us                2019-12-20T13:35:48Z
trafficsplits.split.smi-spec.io               2019-12-20T13:49:11Z
traffictargets.access.smi-spec.io             2019-12-20T13:49:11Z
```

Helm template only produces one crd item.

```shell
helm template maesh/maesh | grep -C 5 traffictargets.access
---
# Source: maesh/templates/smi/smi-access.yaml
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: traffictargets.access.smi-spec.io
spec:
  group: access.smi-spec.io
  version: v1alpha1
  scope: Namespaced
  names:
```

Is anyone able to shine a light on what's actually failing here? My powers of reasoning are failing me completely.

**Comment by iamcryptoki** [comment]
Same issue for me.

**Comment by SantoDE** [comment]
Hey guys,

I'll give it another go and try to reproduce. :)

**Comment by SantoDE** [comment]
Hello all!

I can confirm that behavior and let's say its "intended". That chart is helm v2 compatible. If you try to install it with Helm v3, that issue will happen. Our Helm v3 compatible chart is already in master and will be released with Maesh v1.1.

I'll add a note to the docs for the time in between.

Sorry for the inconveniences!
